src/main/java/io/vertx/ext/mongo/impl/config/SSLSettingsParser.java

```diff
@@ -36,60 +29,55 @@
     this.config = config;
   }
 
-  public SslSettings settings() {
-    final SslSettings.Builder builder = fromConnectionString().orElseGet(this::fromConfiguration);
-    if (config.getBoolean("trustAll", false)) {
-      log.warn("Mongo client has been set to trust ALL certificates, this can open you up to security issues. Make sure you know the risks.");
-      try {
-        final SSLContext context = SSLContext.getInstance("TLS");
-        context.init(null, new TrustManager[]{TrustAllTrustManager.INSTANCE}, new SecureRandom());
-        builder.context(context);
-      } catch (final NoSuchAlgorithmException | KeyManagementException e) {
-        //fail silently on error
-      }
+  public SslSettings settings(Vertx vertx) {
```

Comment:
I'll note that this is a break in a public API, but I suspect this mostly gets used internally.